src/btree/bt_read.c

```diff
@@ -194,8 +200,25 @@ __las_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref)
     locked = false;
 
     /* Check whether the disk image contains all the newest versions of the page. */
-    if (ref->page_las->min_skipped_ts == WT_TS_MAX)
-        return (__instantiate_birthmarks(session, ref));
+    if (page_las->min_skipped_ts == WT_TS_MAX) {
+        WT_RET(__instantiate_birthmarks(session, ref));
+
+        FLD_SET(page->modify->restore_state, WT_PAGE_RS_LOOKASIDE);
+
+        /*
+         * The page image contained the newest versions of data so the updates in lookaside are all
+         * older and we could consider marking it clean (i.e., the next checkpoint can use the
+         * version already on disk).
+         */
+        if (!page_las->has_prepares && !S2C(session)->txn_global.has_stable_timestamp &&
```

Comment:
The conditional is not obvious to me. Could you explain why we're checking this?

Reply:
When the disk image of page contains the newest versions of the page and the data is globally visible, there is nothing required by the checkpoint to take care of it during the checkpoint. So we can mark the page as clean, but only when the page doesn't have prepares and there is no stable timestamp is available. I bring this check from earlier code.